src/client/src/ui/spotTile/components/PriceButton.tsx

```diff
@@ -1,27 +1,57 @@
 import React from 'react'
-import { styled } from 'rt-theme'
+import { styled, Theme } from 'rt-theme'
 import { Direction } from 'rt-types'
+import { keyframes, css } from 'styled-components'
+import { RfqState } from './types'
 
 const hoverColors = {
   [Direction.Buy]: 'blue',
   [Direction.Sell]: 'red',
 }
 
-export const TradeButton = styled.button<{ direction: Direction }>`
+const backgroundEffectKeyframes = ({
+  direction,
+  theme,
+}: {
+  direction: Direction
+  theme: Theme
+}) => keyframes`
+  5% {
+    background-color: ${theme.template[hoverColors[direction]].normal};
+    color: white;
+  }
+  80% {
+    background-color: ${theme.template[hoverColors[direction]].normal};
+    color: white;
+  }
+`
+
+const getAnimationCSSProperty = (props: { direction: Direction; theme: Theme }) => css`
+  animation: ${backgroundEffectKeyframes(props)} 5s;
+`
+
+const backgroundEffect = ({ rfqState, ...rest }: { rfqState: RfqState }) =>
+  rfqState === 'received'
+    ? getAnimationCSSProperty(rest as { direction: Direction; theme: Theme })
+    : ''
+
+export const TradeButton = styled.button<{ direction: Direction; rfqState: RfqState }>`
   background-color: ${({ theme }) => theme.core.lightBackground};
   border-radius: 3px;
+  color: ${({ theme, rfqState, direction }) =>
+    rfqState === 'received' ? theme.template[hoverColors[direction]].normal : 'white'};
```

Comment:
Suggestion: We could use an enum for RfqState instead of a type. That way we can use RfqState.RECEIVED opposed to using a string here.